File: backend/models/notesModel.js
const mongoose = require(`mongoose`);

const Schema = mongoose.Schema;

const NotesSchema = Schema({
  title: {
    type: String,
    required: true,
  },
  semester: {
    type: Number,
    required: true,
  },
  subject: {
    type: String,
    required: true,
  },
  module: {
    type: String,
    required: true,
  },
  file: {
    type: String,
    required: true,
  },
});

module.exports = mongoose.model("Note", NotesSchema);

File: go-backend/model/Notesmodel.go
package model

type Note struct {
	Title       string `json:"title,omitempty" bson:"title,omitempty"`
	Semester    string `json:"semester,omitempty" bson:"semester,omitempty"`
	SubjectCode string `json:"sunjectCode,omitempty" bson:"subjectCode,omitempty"`
	Module      string `json:"module,omitempty" bson:"module,omitempty"`
	File        string `json:"file,omitempty" bson:"file,omitempty"`
}
